File: connector-service/src/core/use_cases/AddExpenses.ts
import {IExpensesService} from "../domain/expenses/IExpensesService";
import {TelegramId} from "../domain/TelegramId";
import {Message} from "../domain/Message";
import {Expenses} from "../domain/expenses/Expenses";
import {ResponseChat} from "./ResponseChat";
import {format} from "./Utils";

export class AddExpenses {
    constructor(private readonly expensesService: IExpensesService) {
    }

    async execute(telegramId: TelegramId, message: Message): Promise<ResponseChat> {
        return this.toResponseChat(await this.expensesService.add(telegramId, message));
    }

    toResponseChat(expenses: Expenses): ResponseChat {
        return  {
            message: `${format(expenses.category)} expenses added ✅`
        }
    }
}

File: bot-service/src/core/use_cases/add_expenses.py
from src.core.domain.expenses import Expenses, ExpensesService
from src.core.domain.user import UsersRepository


class AddExpenses:
    def __init__(self, users_repository: UsersRepository,
                 expenses_service: ExpensesService):
        self._users_repository = users_repository
        self._expenses_service = expenses_service

    def execute(self, telegram_id: str, message: str) -> Expenses:
        user = self._users_repository.get(telegram_id)
        expenses = self._expenses_service.create(message)

        user.add_expenses(expenses)
        self._users_repository.save(user)

        return expenses
